### ISSUE TITLE
Use other terminal app on Mac OS X?

### ISSUE THREAD

**Issue by tobiasger** [wbond/sublime_terminal#174]
I'm mostly using [Hyper](https://hyper.is/) instead of the built-in terminal app in Mac OSX and I'm looking for a way to use sublime_terminal with Hyper. There doesn't seem to be a way of changing the app being used for the plugin at the current version (or at least nothing else besides iTerm). Is this something that you are thinking of adding support for later?

**Comment by twolfson** [comment]
Sorry for the slow reply, I was on an off-the-grid vacation. I think Hyper has enough traction behind it that we could add a `.sh` script for it. Would you mind writing one (should be similar to copy/paste `iTerm.sh`)?

**Comment by tobiasger** [comment]
Unfortunately I'm not familiar with `.sh` script... Let's hope that any other Hyper user would care to make one. :)

**Comment by aytee17** [comment]
Electron doesn't support AppleScript which is used to launch the terminal. electron/electron#4418

In the mean time I wrote a hacky solution which involves launching Hyper and then inputting commands via keyboard events.

```
#!/bin/bash

CD_CMD="cd "\\\"$(pwd)\\\"" && clear"

osascript &>/dev/null <<EOF
	tell application "Hyper"
		activate
		tell application "System Events"
			key code 17 using command down   # Open new tab
			delay 0.5                        # Wait a bit for the tab to open
			keystroke "$CD_CMD"              # Input command
			key code 36  #(presses enter)    # Press enter
		end tell
	end tell
EOF
```

**Comment by infn8** [comment]
So i have made a _very_ simple hyper script but I can't seem to get it working with params.  can params in this package be passed to other scripts?  

ok so here is the deal.  Installing hyper adds a command called `hyper` so that calling

```bash
which hyper
```
for me yeilds `/usr/local/bin/hyper` and when you do `hyper .` in any folder it opens a new hyper tab to that folder `hyper /Applications` will run hyper and immediately do `cd /Applications`. ok so with all of that i build this script:

```bash
#!/bin/bash

hyper $1
```

that's it.  it just calls hyper with param # 1

then `cp ./hyperopen.sh /usr/local/bin/hyperopen` adds it to my path and now anywhere in terminal i can write `hyperopen /Applications` and hyper opens to the Applications folder so that's cool. now i added this in my `~/Library/Application Support/Sublime Text 3/Packages/User/Terminal.sublime-settings`:

```json
{
    "terminal": "hyperopen",
    "parameters": ["%CWD%"]
}
```
but nothing.  it runs hyperopen and hyper opens but there is no param passed and hyper does not attemp to `cd` anywhere.

Can anyone confirm if that param is getting sent or what?  Thanks.

![Help me you're my only hope](https://media.giphy.com/media/l1AsUvwEEBorHpMpG/giphy.gif "Help Me you're my only hope")

**Comment by twolfson** [comment]
Those `terminal` and `parameters` look right to me. Can you verify that you accessed the `.sublime-settings` file via:

```
Preferences > Package Settings > Terminal > Settings – User
```

Aside from that, try debugging it to make sure the `hyperopen` in your `$PATH` is respecting the parameters:

```js
{
    "terminal": "hyperopen",
    "parameters": ["/Applications"]
}
```

If that persists, try launching in the Sublime Text console (ctrl+`) without our Terminal package:

```python
subprocess.Popen(["hyperopen, "/Applications")
```

**Comment by infn8** [comment]
It actually seems to work now.  I think I needed a system reset or something.

**Comment by twolfson** [comment]
Great! Glad to hear your problem has been resolved =)

**Comment by twolfson** [comment]
We landed the workaround AppleScript in #205. I forgot about this issue before landing but I'm guessing `hyper $1` doesn't necessarily open a new tab, is that right?
